### PR TITLE
feat(html): Add BreakdownTable.ts

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [x] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

**Comment by jbrundage** [comment]
@GordonSmith this is ready to be reviewed

File: packages/html/src/SimpleTable.ts [hunk 1]
```diff
@@ -46,6 +48,10 @@ export class SimpleTable extends HTMLWidget {
             ;
         trSelection.exit().remove();
     }
+
+    transformData(data) {
+        return data;
+    }
```
Comment (GordonSmith) [inline packages/html/src/SimpleTable.ts]: Mark as protected and move above the enter method and change its signature to:
```
    transformedData() {
        return this.data();
    }
```

File: packages/html/src/SimpleTable.ts [hunk 2]
```diff
@@ -28,11 +30,11 @@ export class SimpleTable extends HTMLWidget {
             .text(_d => (_d).toString())
             ;
         theadTrSelection.exit().remove();
-        const trSelection = this._tbody.selectAll("tr").data(this.data());
+        const trSelection = this._tbody.selectAll("tr").data(this.transformData(this.data()));
```
Comment (GordonSmith) [inline packages/html/src/SimpleTable.ts]: Don't pass data into it.

File: packages/html/src/BreakdownTable.ts [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+                        width:${Math.floor(99 / colCount)}%;
+                    ">${row[0]}: ${row[1]}</div>`
+                    ).join("")
+                    }</div>`;
+            })
+            ;
+        this.transformData = function () {
+            const rowCount = this.useCalculatedRowCount() ? this.calculateRowCount() : this.rowCount();
+            return this.breakdownData(rowCount);
+        };
```
Comment (GordonSmith) [inline packages/html/src/BreakdownTable.ts]: This shoud be a regular method:

```
transformedData() {
    const rowCount = this.useCalculatedRowCount() ? this.calculateRowCount() : this.rowCount();
    return this.breakdownData(rowCount);
}
```

File: packages/html/src/BreakdownTable.ts [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    protected _tbody;
+    protected _tooltip;
+    constructor() {
+        super();
+    }
+
+    protected transformData() {
+        const rowCount = this.useCalculatedRowCount() ? this.calculateRowCount() : this.rowCount();
+        return this.breakdownData(rowCount);
+    };
```
Comment (GordonSmith) [inline packages/html/src/BreakdownTable.ts]: Remove ";"

**Pull request merged**